test/integration/listener_extension_discovery_integration_test.cc

```diff
@@ -24,9 +24,11 @@ class ListenerExtensionDiscoveryIntegrationTest : public Grpc::GrpcClientIntegra
 
   void addDynamicFilter(const std::string& name, bool apply_without_warming,
                         bool set_default_config = true, bool rate_limit = false,
-                        ListenerMatcherType matcher = ListenerMatcherType::NULLMATCHER) {
+                        ListenerMatcherType matcher = ListenerMatcherType::NULLMATCHER,
+                        const std::string& cluster_name = "ecds_cluster",
+                        bool second_stream = false) {
```

Comment:
High level comment: are these 2 parameters highly coupled? Looking at the rest of the code, it seems that `cluster_name` is "ecds_cluster iff `second_stream` is false.
If so, the functions here should be more generic, i.e., passing an index of the ECDS connection/stream/cluster_name.

Might also want to take a look at "scoped_rds.h", "listener_lds_integration_test.cc" and "leds_integration_test.cc", and the way they use `stream_by_resource_name_`.

Reply:
Thanks for the comments!  Done.
In this case, using the boolean second_stream can derive the cluster_name.